### PR TITLE
fix for latest deal.II master: support cell_weight

### PR DESCRIPTION
**Pull request by tjhei**

### CODE REVIEW

**Comment by tjhei** [comment]
needs to go into 2.5

File: source/particle/world.cc [hunk 1]
```diff
@@ -354,6 +365,20 @@ namespace aspect
...
+            n_particles_in_cell = particle_handler->n_particles_in_cell(cell);
+            break;
+
+          case CellStatus::cell_invalid:
+            break;
+
+          case CellStatus::children_will_be_coarsened:
+            for (const auto &child : cell->child_iterators())
+              n_particles_in_cell += particle_handler->n_particles_in_cell(child);
+            break;
```
Comment (tamiko) [inline source/particle/world.cc:380]: Same here. I don't think it would hurt to make the names available again via using declarations or similar trickery.

**Comment by bangerth** [comment]
@tamiko That will provide a temporary fix that would have avoided the immediate breakage. But in the long run, @tjhei's changes are necessary anyway.

**Comment by tjhei** [comment]
It's still necessary to avoid deprecation warnings.

**Pull request merged**